src/main/kotlin/org/rust/lang/core/resolve/LangItems.kt

```diff
@@ -35,13 +36,39 @@ fun findIteratorItemType(project: Project, ty: RustType): RustType {
     return rawType.substitute(typeParameterMap)
 }
 
+fun findIndexOutputType(project: Project, containerType: RustType, indexExprType: RustType): RustType {
+    val impls = RsImplIndex.findImpls(project, containerType)
+    for (impl in impls) {
+        val trait = impl.traitRef?.resolveToTrait ?: continue
+        if (trait.isIndex) {
+            // TODO: get index type from impl declaration
+            val indexType = impl.functionList
```

Comment:
But the declaration of output type doesn't contain type of possible index expression. Or do you suggest it instead of using `lookupAssociatedType` function?